guides/dsl.md

```diff
@@ -0,0 +1,241 @@
+    () => { west.firstName; return `${{west.firstName}} ${{west.lastName}}`},
+  );
+  return [
+    record("html/element", {tagname: "div", text: fullName}),
+  ];
+});
+```
+
+### Functions
+
+The standard library in Eve has been redone in the DSL. To use library functions, you must now bring in "lib" explicitly when defining your block. From lib you can access the various standard library functions supported by the runtime so far. 
+
+```javascript
+program.block("display the student's full names", ({find, record, lib}) => {
+  find("angle" degrees)
+  let result = math.sin(degrees)
+  return [
+    record("html/element", {tagname: "div", text: result})
+  ];
+});
+```
+
+for now you can find a list of functions in [src/runtime/stdlib.ts](https://github.com/witheve/Eve/blob/refactor-editor/src/runtime/stdlib.ts)
+
+The interface for wrapping functions for use within Eve is also revamped for the new runtime. When writing a function wrapper, you must ensure that the function meets the following qualifications
+
+1. The function must be referentially transparent, meaning given the same input, the function should return the same output.
+
+2. The function must be monotonic, meaning that additional inputs should never decrease the number of outputs.
+
+If the function meets these two qualifications, it can be wrapped using `makeFunction()` e.g.:
+
+```javascript
+makeFunction({
+  name: "math/sin",             
+  args: {a: "number"},
+  returns: {result: "number"},
+  apply: (a:number) => {
+    return [Math.sin(a/180 * Math.PI)];
+  }
+});
+```
+
+## Importing and Exporting Records 
+
+### Getting Data Into Eve - `inputEavs()`
+
+You can import raw EAVs into Eve with the `inputEavs()` function. Due to limitations of the host language, in order for it to work correctly, `inputEavs()` can only be used after you define all blocks.
+
+`inputEavs()` takes as input a list of entity, attribute, value triples. The entity value identifies the record to which the attribute and value belong, so it must be unique to that record. For example:
+
+```javascript
+prog.inputEavs([[0,"tag","person"],[0,"name","Archibald"])
+```
+
+This will create a record tagged "person" with the name attribute "Archibald".
+
+### Getting Records out of Eve - Watchers
+
+Watchers are a new feature native to the DSL that allow you to monitor changes in specific records and react to them with a callback function. Like blocks, watchers also attach to a `program`.
+
+```javascript
+  program.watch("Export information about students", ({find, lookup, record}) => {
+    // search for records tagged student
+    let student = find("student");
+    // lookup attributes and values related to each student
+    let {attribute, value} = lookup(student);
+    return [
+      // Add these attributes and values to the student, which creates a diff to which we can react
+      student.add(attribute, value)
+    ];
+  })
+  // React to each addition or removal
+  .asDiffs((diff) => {
```

Comment:
Need a section on `asObjects` as well, since people will probably want to use that one more often than asDiffs